### PR TITLE
feat(toast): add notice variant (FE-4402)

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour

Toast component has the alternative style variant, which is always full width and rendered in the bottom of the screen.
Toast with that variant should animate from the bottom.
Alternative style can choose any content, icon/pill/text etc, default just text.
Alternative style can be dismissible or delay to animate out.

image

### Checklist

- [x] Commits follow our style guide
~~- [ ] Related issues linked in commit messages if required~~
- [x] Screenshots are included in the PR if useful
~~- [ ] All themes are supported if required~~
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
~~- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required~~
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/toast/toast.spec.js [hunk 1]
```diff
@@ -314,6 +320,39 @@ describe("Toast", () => {
...
+      expect(wrapper.find(ToastWrapper).props().notice).toBe(true);
+    });
+
+    it("then the prop should be passed to ToastContentStyle", () => {
+      expect(wrapper.find(ToastContentStyle).props().notice).toBe(true);
+    });
+
+    it("then the TypeIcon should not be rendered", () => {
+      expect(wrapper.find(TypeIcon).exists()).toBe(false);
+    });
```
[the remark this answers is not in the document]
Comment (grabkowski) [reply src/components/toast/toast.spec.js:354]: just code coverage, chromatic does all the job, but we have to reach all the conditionals

**Review by edleeks87 — DISMISSED** [review]

**Review by edleeks87 — COMMENTED** [review]
Happy to approve when @mkrds' comment has been addressed

**Review by mkrds — APPROVED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Comment by harpalsingh** [comment]
Just noticed the border on this variant @grabkowski 

Screenshot 2022-09-29 at 09 13 40

This wasnt in the design spec, so checking if it was added/missed for any reason?

**Review by edleeks87 — DISMISSED** [review]

**Review by mkrds — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Pull request merged**